algoliasearch/src/main/java/com/algolia/search/saas/BaseAPIClient.java

```diff
@@ -525,7 +525,13 @@ private synchronized JSONObject _request(Method m, String url, String json, List
                 throw new AlgoliaException(message);
             } else {
                 try {
-                    errors.put(host, _toCharArray(stream));
+                    final String errorMessage;
+                    if (stream == null) {
+                        errorMessage = "InputStream is null: No connection/error while connecting/no error data";
+                    } else {
+                        errorMessage = _toCharArray(stream);
+                    }
+                    errors.put(host, errorMessage);
                 } catch (IOException e) {
```

Comment:
We need to handle a `null` stream everywhere: same logic must apply a few lines above.